### DISCUSSION TITLE
Including Ui Frameworks

### DISCUSSION THREAD

**Discussion by Leviathan91** [vercel/next.js#15051] [Help]
Hi everyone,
I want to build some websites ( starting with a own blog, that's the easiest I guess) with NextJS, because I think it's features are great.
I come more from a backend-background, currenctly working on my front-end skills but well, they are..quite limited.

So I thought to get something bootstrapped quickly, I'd use a UI Framework to help me out.

My concern is the following:
How do I include these libraries/frameworks with NextJS properly?
For example this [Paper Kit React](https://demos.creative-tim.com/paper-kit-react/#/index).  
In its [documentation](https://demos.creative-tim.com/paper-kit-react/#/documentation/introduction) it is shown that you have to load their entire CSS. 
Does NextJS SSR still dismiss all the unused CSS in production?  Because if it doesn't, using such a framework would actually ruin the benefits on NextJS.

So I guess my questions are:

- Can I use UI frameworks/templates with NextJS and still gain all of NextJS benefits? How to do so?

And if I can't, my follow-up would be:

-  What would you recommend me to do? I feel like writing everything from scratch seems like a lot of work - especially regarding the Responsive Aspect. Are there NextJS best practices/recommended paths for a fullstack Project?

Thank you for your time

**Comment by leerob** [comment]
> Can I use UI frameworks/templates with NextJS and still gain all of NextJS benefits? How to do so?

Yes, absolutely! I would recommend checking out a few of these examples.

- [with-ant-design](https://github.com/vercel/next.js/tree/canary/examples/with-ant-design)
- [with-chakra-ui](https://github.com/vercel/next.js/tree/canary/examples/with-chakra-ui)
- [with-material-ui](https://github.com/vercel/next.js/tree/canary/examples/with-material-ui)
- [with-react-bootstrap](https://github.com/vercel/next.js/tree/canary/examples/with-react-bootstrap)
- [with-tailwindcss](https://github.com/vercel/next.js/tree/canary/examples/with-tailwindcss)

> What would you recommend me to do? I feel like writing everything from scratch seems like a lot of work - especially regarding the Responsive Aspect. Are there NextJS best practices/recommended paths for a fullstack Project?

Personally, I'm a big fan on [Chakra UI](https://chakra-ui.com/). [Tailwind CSS](https://tailwindcss.com/) is another popular option if you don't want CSS-in-JS. I'd give the docs a read and learn more about both!
